### PR TITLE
Added picard addOrReplaceReadgroup

### PR DESCRIPTION
**Pull request by J35P312**
modified:   config/run_test_files.txt
	new file:   picardtools_addorreplacereadgroups.t

### CODE REVIEW

File: lib/MIP/Program/Alignment/Picardtools.pm [hunk 1]
```diff
@@ -32,12 +32,173 @@ BEGIN {
...
 
 ## Constants
 Readonly my $SPACE => q{ };
 
+sub picardtools_addorreplacereadgroups {
+## Function : Perl wrapper for writing picardtools addorreplacereadgroups recipe to $FILEHANDLE. Based on picardtools 2.5.0.
+## Returns  : @commands
+## Arguments: $infile_path                    => Infile paths
+##          : $outfile_path                   => Outfile path
+##          : $stderrfile_path                => Stderrfile path
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Alignment/Picardtools.pm]: Order in Alphabetically

File: lib/MIP/Program/Alignment/Picardtools.pm [hunk 2]
```diff
@@ -32,12 +32,173 @@ BEGIN {
...
+    my $java_use_large_pages;
+    my $readgroup_id;
+    my $readgroup_library;
+    my $readgroup_platform;
+    my $readgroup_platform_unit;
+
+    my $tmpl = {
+        FILEHANDLE  => { store => \$FILEHANDLE },
+        infile_path => {
+            required    => 1,
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Alignment/Picardtools.pm]: Order keys AB

File: lib/MIP/Program/Alignment/Picardtools.pm [hunk 3]
```diff
@@ -32,12 +32,173 @@ BEGIN {
...
+            defined     => 1,
+            strict_type => 1,
+            store       => \$infile_path
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Alignment/Picardtools.pm]: Add comma to last element in key value pair
\$infile_path -> \$infile_path,

File: lib/MIP/Program/Alignment/Picardtools.pm [hunk 4]
```diff
@@ -32,12 +32,173 @@ BEGIN {
...
+
+    ## Return java core commands
+    if ($java_jar) {
+
+        @commands = java_core(
+            {
+                memory_allocation    => $memory_allocation,
+                java_use_large_pages => $java_use_large_pages,
+                temp_directory       => $temp_directory,
+                java_jar             => $java_jar,
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Alignment/Picardtools.pm]: Order AB

**Comment by henrikstranneheim** [comment]
@J35P312 Please remove the sir t/config/ it has become redundant

File: lib/MIP/Program/Alignment/Picardtools.pm [hunk 5]
```diff
@@ -32,12 +32,173 @@ BEGIN {
...
+            store       => \$java_use_large_pages,
+            strict_type => 1,
+        },
+        outfile_path => {
+            defined     => 1,
+            required    => 1,
+            strict_type => 1,
+            store       => \$outfile_path,
+        },
+        memory_allocation => { strict_type => 1, store => \$memory_allocation },
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Alignment/Picardtools.pm]: Order this AB

**Comment by J35P312** [comment]
Hello!
Now all should be fixed!

**Comment by henrikstranneheim** [comment]
👍

**Pull request merged**